camunda-bpm-camel-common/src/main/java/org/camunda/bpm/camel/component/producer/StartProcessProducer.java

```diff
@@ -76,7 +90,63 @@ public void process(Exchange exchange) throws Exception {
 
     exchange.setProperty(EXCHANGE_HEADER_PROCESS_DEFINITION_ID, instance.getProcessDefinitionId());
     exchange.setProperty(EXCHANGE_HEADER_PROCESS_INSTANCE_ID, instance.getProcessInstanceId());
-    exchange.getOut().setBody(instance.getProcessInstanceId());
+    
+    setOutBody(exchange, instance);
+    
+  }
+  
+  private void setOutBody(final Exchange exchange, final ProcessInstance instance) {
+    if (parameters.containsKey(COPY_PROCESS_VARIABLES_TO_OUT_BODY_PARAMETER)) {
+      if (historyService == null) {
+          throw new RuntimeCamelException("Fetching process instance variables is not supported "
+                                          + "because a Camunda history service is needed but not available!");
+      }
+      
+      final String variableName = parameters.get(COPY_PROCESS_VARIABLES_TO_OUT_BODY_PARAMETER).toString();
+      
+      if (variableName.equals("*")) {
+        final List<HistoricVariableInstance> variables = historyService.createHistoricVariableInstanceQuery()
+                    .processInstanceIdIn(instance.getProcessInstanceId())
+                    .list();
```

Comment:
When changing variables, this should result in the different versions of the variable instance - which is not what you want I think

Reply:
You are right. I'll check this.

Reply:
In my understanding it returns only the last value of a variable, as it returns always at the first wait state of the process - isn't it?